.github/workflows/github-pipelines.yml

```
@@ -15,14 +15,32 @@ jobs:
         fetch-depth: 5  # Fetch 5 commits worth of history
 
-    - name: Install Git and Curl
+    - name: Set up Git and Curl
       run: |
         sudo apt-get update
         sudo apt-get install -y git curl
         git --version  # Verify Git installation
         curl --version # Verify curl installation
 
+    - name: Cache dependencies (Git & Curl)
+      uses: actions/cache@v2
```

actionlint (1.7.4)

the runner of "actions/cache@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)